stl/inc/utility

```diff
@@ -441,21 +441,17 @@ struct pair { // store a pair of values
     _CONSTEXPR20 void swap(pair& _Right) noexcept(
         _Is_nothrow_swappable<_Ty1>::value && _Is_nothrow_swappable<_Ty2>::value) {
         using _STD swap;
-        if (this != _STD addressof(_Right)) {
-            swap(first, _Right.first); // intentional ADL
-            swap(second, _Right.second); // intentional ADL
-        }
+        swap(first, _Right.first); // intentional ADL
+        swap(second, _Right.second); // intentional ADL
     }
 
 #if _HAS_CXX23
     template <int = 0> // see GH-3013
     constexpr void swap(const pair& _Right) const
         noexcept(is_nothrow_swappable_v<const _Ty1> && is_nothrow_swappable_v<const _Ty2>) {
         using _STD swap;
-        if (this != _STD addressof(_Right)) {
-            swap(first, _Right.first); // intentional ADL
-            swap(second, _Right.second); // intentional ADL
-        }
+        swap(first, _Right.first); // intentional ADL
+        swap(second, _Right.second); // intentional ADL

```

Comment:
I think we can test this overload with `pair<swap_counter&, tuple<>>` or `pair<swap_counter&, SomeOtherType&>`. Such types are const-swappable since C++23.

Reply:
This is a test for this issue. If we are going to test these things, would it be better to change the name of the test?
Or leave it to their own tests?

Reply:
IMO they are all about this issue.

Reply:
Done